app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -0,0 +1,240 @@
+require 'azure_mgmt_compute'
+require 'azure_mgmt_resources'
+require 'azure_mgmt_network'
+require 'azure_mgmt_storage'
+
+module ForemanAzureRM
+  class AzureRMCompute
+    attr_accessor :sdk
+    attr_accessor :azure_vm
+
+    Compute = Azure::Compute::Profiles::Latest::Mgmt
+    ComputeModels = Compute::Models
+
+    def initialize(azure_vm: ComputeModels::VirtualMachine.new, sdk: sdk)
+      @azure_vm = azure_vm
+      @sdk = sdk
+    end
+
+    def id
+      @azure_vm.id
+    end
+
+    def name
+      @azure_vm.name
+    end
+
+    def persisted?
+      !!identity && !!id
+    end
+
+    def region
+      @azure_vm.location
+    end
+
+    def region=(setregion)
+      @azure_vm.location = setregion
+    end
+
+    def resource_group
+      @azure_vm.resource_group
+    end
+
+    def resource_group=(setrg)
+      @azure_vm.resource_group = setrg
+    end
+
+    def vm_size
+    end
+
+    def vm_size=(setvsize)
+      @azure_vm.hardware_profile.vm_size = setvsize
```

Comment:
We probably need to initialize `hardware_profile` in case it was not initialized before . Especially when setting up an empty instance.